ParserHelpers.h

```diff
@@ -14,6 +15,10 @@ void ignoreItem(const std::string& unused, std::istream& theStream);
 void ignoreObject(const std::string& unused, std::istream& theStream);
 void ignoreString(const std::string& unused, std::istream& theStream);
 
+template <typename T>
+[[nodiscard]] std::enable_if_t<std::is_integral_v<T>, T> stringToInteger(const std::string& str, bool skipPartialMatchWarning = false);
```

Comment:
I don't know enough about templates, but is it possible to not have the std::enable_if_t part in the header? It makes it hard to read and see what's happening overall.

Reply:
I don't know of a way, but I've replaced
```std::enable_if_t<std::is_integral_v<T>, T>```
with
```std::enable_if_t<std::numeric_limits<T>::is_integer, T>```
and added a comment
```// function template only enabled for integer types```
to make it somewhat easier to understand.